common_package_py/commands.py

```diff
@@ -0,0 +1,344 @@
+        min_val (float | None, optional): Minimum value allowed for numeric keys. Defaults to None.
+        max_val (float | None, optional): Maximum value allowed for numeric keys. Defaults to None.
+    """
+
+    def __init__(self, required: bool, data_types: set[data_type] | data_type, min_val: float | None = None, max_val: float | None = None):
+        self.required = required
+
+        # Check if data_types is a single data_type and convert it to a set
+        if isinstance(data_types, data_type):
+            data_types = {data_types}
+
+        # Check if data_types is a set
+        if not isinstance(data_types, set):
+            raise RuntimeError(
+                f"JsonKeyDefinition::JsonKeyDefinition: data_types has the wrong type: {type(data_types)}")
+
+        # Check if each element in data_types is of type data_type
+        for dt in data_types:
+            if not isinstance(dt, data_type):
+                raise RuntimeError(
+                    f"JsonKeyDefinition::JsonKeyDefinition: data_type '{dt}' has the wrong type: {type(dt)}. Expected: {type(data_type)}")
+
+        # Check if data_types is not empty
+        if len(data_types) <= 0:
+            raise RuntimeError(
+                "JsonKeyDefinition::JsonKeyDefinition: data_types is empty")
+
+        # Set the data_types attribute
+        self.data_types = data_types
+
+        # Check if min_val and max_val are provided and swap them if min_val > max_val
+        if min_val is not None and max_val is not None and min_val > max_val:
+            self.min_val, self.max_val = max_val, min_val
+        else:
+            self.min_val, self.max_val = min_val, max_val
+
+    @staticmethod
+    def data_type_to_string(data_type: data_type) -> str:
+        """
+        Converts a data type to its string representation.
+
+        Args:
+            data_type (data_type): The data type to convert.
+
+        Returns:
+            str: The string representation of the data type.
+        """
+
+        return data_type.name.lower()
+
+    def check_bounds(self, json_iter: any) -> bool:
+        """
+        Checks if the value of the JSON key is within the specified bounds.
+        The bounds are defined by the minimum and maximum values allowed for the
+        supported data types. Only specified bounds are checked. If no bounds are
+        specified, all values will be accepted. Every data type that is not of
+        any `number*` data type will always return true.
+
+        Note:
+            Also returns true if the json_iter has a type different from `number*`
+
+        Args:
+            json_iter (any): The value of the JSON key.
+
+        Returns:
+            bool: True if the value is within the bounds, False otherwise.
+        """
+
+        jsk_type_check = JsonKeyDefinition(False, {data_type.NUMBER})
+
+        if data_type.NUMBER in self.data_types or \
```

Comment:
Schnittmenge von sets nutzen:
```suggestion
        if {data_type.NUMBER, data_type.NUMBER_FLOAT,  data_type.NUMBER_INTEGER, data_type.NUMBER_UNSIGNED} & self.data_types:
```
Oder expliziter:
```suggestion
        if 0 < len({data_type.NUMBER, data_type.NUMBER_FLOAT,  data_type.NUMBER_INTEGER, data_type.NUMBER_UNSIGNED} & self.data_types) :
```
Siehe auch hier: https://www.online-python.com/mMO5LqwkPc

Reply:
Macht meiner Meinung nach schlechter lesbar bei keinem Vorteil.

Reply:
Das ist aber mehr pythonic.
Und außerdem wollen wir den zukünftigen Entwicklern ja ne challenge bieten.

Reply:
Welche zukünftigen Entwickler? Das landet danach in der Versenkung, wie alles andere auch bisher.

Reply:
Hallo, das wird später sogar von der NASA verwendet werden.